### PR TITLE
Enable database in tests and fix updating matches

### PR DESCRIPTION
**Pull request by marcocastignoli**
- Enable database check in the tests: assert that the match exists in sourcify_matches with address,chain,status
- There was a problem while updating a match from partial to perfect, it didn't update the row in sourcify_matches, this PR solves this problem
- I had to make `assertVerification` and `assertVerificationSession` async. So then I had to update all the calls in the tests

### TODO

- [x] After we merge #1298 I'm going to update this with new logging

### CODE REVIEW

**Comment by marcocastignoli** [comment]
For some reason, tests are failing on CircleCI while they are successfully running  locally.

**Comment by marcocastignoli** [comment]
I thought that the env variables in `services/server/test/server.js` were enough in order to start Sourcify with specific env variables. It turns out that is not the case, I fixed this by passing the environments in circleci's `npx lerna run build && npx lerna run lerna-test` configuration

File: services/server/src/server/services/storageServices/SourcifyDatabaseService.ts [hunk 1]
```diff
@@ -144,13 +144,25 @@ export class SourcifyDatabaseService
         creationMatch: match.creationMatch,
       });
     } else if (type === "update") {
-      if (verifiedContractId) {
-        await Database.updateSourcifyMatch(this.databasePool, {
+      if (!verifiedContractId) {
+        throw new Error(
```
Comment (marcocastignoli) [inline services/server/src/server/services/storageServices/SourcifyDatabaseService.ts:148]: @kuzdogan do we need to add a logger.error() before throwing?

Comment (kuzdogan) [reply services/server/src/server/services/storageServices/SourcifyDatabaseService.ts:148]: Anyting we want to be alerted about needs to be `logger.error` logged. If that's the case yes.

**Comment by marcocastignoli** [comment]
@kuzdogan All the errors thrown into storage services are logged inside each catch

e.g.
https://github.com/ethereum/sourcify/blob/a6c54f6f72e310402e5e87e0b57b45c448722578/services/server/src/server/services/StorageService.ts#L166-L174

So I don't think additional error-related logs are necessary, lmk what you think.

**Comment by kuzdogan** [comment]
@marcocastignoli If that is the case, ok this way is the way to go

**Review by kuzdogan — APPROVED** [review]

**Pull request merged**